fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/APIVersion.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * APIVersion.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2022 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.provider.foundationdb;
+
+import com.apple.foundationdb.annotation.API;
+import com.apple.foundationdb.record.RecordCoreArgumentException;
+
+import javax.annotation.Nonnull;
+
+/**
+ * An enum representing the different supported FDB API versions. The
+ * <a href="https://apple.github.io/foundationdb/api-general.html#api-versions">FDB API version</a>
+ * controls the minimum FDB server version that can be successfully connected to, as well as
+ * what features are supported by the FDB client. It is therefore important that in scenarios where
+ * there are FDB servers on different versions that the API version be configured to not exceed the
+ * lowest FDB server version. If the API version is set too high, the FDB client may hang when
+ * trying to connect to the server.
+ *
+ * <p>
+ * Because FDB API versions also control which features are guaranteed to be supported, there may
+ * be Record Layer features that are gated on specific FDB API versions. For that reason, if there
+ * is a new feature that relies on specific FDB server features being available, the client may
+ * need to be configured with the appropriate API version before that feature can be used.
+ * </p>
+ *
+ * <p>
+ * A note on Record Layer API stability: the elements of this enum are expected to shift as support
+ * for FDB versions are deprecated and removed. As a result, the elements of this enum will be added
+ * and removed as support for old FDB versions is dropped and new FDB versions is added.
+ * </p>
+ *
+ * @see FDBDatabaseFactory#setAPIVersion(APIVersion)
+ */
+@API(API.Status.MAINTAINED)
+public enum APIVersion {
+    /**
+     * API version corresponding to FoundationDB 6.3.
+     */
+    API_VERSION_6_3(630),
+    /**
+     * API version corresponding to FoundationDB 7.0.
+     */
+    API_VERSION_7_0(700),
```

Comment:
Record layer hasn't been tested with FDB 700, can there be any risks by setting the API_VERSION to this?

Reply:
I think the risks of setting the API version to 700 and to 710 are about the same, which is that we don't (currently) run our PRBs at any API version other than 630. There were API differences between 630 and 700, though not for features that seemed relevant to the Record Layer's usage of the FDB API: https://apple.github.io/foundationdb/api-version-upgrade-guide.html#api-version-700

I could see the argument that we should mark `.setAPIVersion` as `EXPERIMENTAL`, and maybe the same for this enum, for at least until we have a more robust test story for supporting multiple API versions.

Reply:
Multiple PRBs (per FDB version) seem like a good (and not expensive) way to resolve this. FDB versions are not so frequently adopted to make this a time sink for the development effort.

Reply:
Okay, I've marked the new APIs as EXPERIMENTAL and noted that that's at least partially because we want some better testing of non-default API versions